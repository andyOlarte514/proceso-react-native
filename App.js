/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {useState} from 'react';
import {
  SafeAreaView,
  TextInput,
  Text,
  View,
  Button,
  StyleSheet,
} from 'react-native';

import {validateText} from './utils.js';

const App = () => {
  const [inputValue, setInputValue] = useState('');
  const [messageTextBalanced, setMessageTextBalanced] = useState('');

  const handleValidateText = () => {
    return validateText(inputValue)
      ? setMessageTextBalanced(
          'Los parentesis, corchetes, etc. en el STRING estan equilibrados',
        )
      : setMessageTextBalanced(
          'Los parentesis, corchetes, etc. en el STRING NO estan equilibrados',
        );
  };

  return (
    <SafeAreaView>
      <View style={styles.container}>
        <TextInput
          style={styles.input}
          value={inputValue}
          onChangeText={value => setInputValue(value)}
        />
        <Button
          disabled={!inputValue}
          title="Validate Text"
          onPress={handleValidateText}
        />
        {inputValue && messageTextBalanced && (
          <Text>{messageTextBalanced}</Text>
        )}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingVertical: 30,
    paddingHorizontal: 30,
  },
  input: {
    borderWidth: 2,
    borderColor: '#808080',
    marginBottom: 5,
  },
});

export default App;
