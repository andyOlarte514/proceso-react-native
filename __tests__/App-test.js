/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import {render, fireEvent, waitFor} from '@testing-library/react-native';

describe('App principal screen', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Press 'Validate Text' button when i add wrong text", async () => {
    const {getByText, getByDisplayValue} = render(<App />);
    const input = getByDisplayValue('');
    const fireEventButtonValidateText = getByText('Validate Text');
    fireEvent.changeText(input, '(texto de prueba');
    fireEvent.press(fireEventButtonValidateText);
    await waitFor(() =>
      expect(
        getByText(
          'Los parentesis, corchetes, etc. en el STRING NO estan equilibrados',
        ),
      ).toBeTruthy(),
    );
  });

  it("Press 'Validate Text' button when i add correct text", async () => {
    const {getByText, getByDisplayValue} = render(<App />);
    const input = getByDisplayValue('');
    const fireEventButtonValidateText = getByText('Validate Text');
    fireEvent.changeText(input, '(texto de prueba)');
    fireEvent.press(fireEventButtonValidateText);
    await waitFor(() =>
      expect(
        getByText(
          'Los parentesis, corchetes, etc. en el STRING estan equilibrados',
        ),
      ).toBeTruthy(),
    );
  });
});
