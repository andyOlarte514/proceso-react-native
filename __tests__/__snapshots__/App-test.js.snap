// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App principal screen renders correctly 1`] = `
<RCTSafeAreaView>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
        "paddingVertical": 30,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#808080",
          "borderWidth": 2,
          "marginBottom": 5,
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Validate Text
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
